### PR TITLE
Alternative support for VS2015 C runtime

### PR DESCRIPTION
**Pull request by rainers**
This uses "weak" linkage to select code path depending on MS C runtime version. It doesn't need additional objects on the comand line, but legacy_stdio_definitions.lib for VS2015.

### CODE REVIEW

**Comment by MartinNowak** [comment]
Nice

**Comment by kinke** [comment]
Continued here: https://github.com/D-Programming-Language/druntime/pull/1361

File: src/rt/stdio_msvc.c [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        stdin = __acrt_iob_func(0);
+        stdout = __acrt_iob_func(1);
+        stderr = __acrt_iob_func(2);
+    }
+    else if (&__iob_func != (void*) &_nullfunc)
+    {
+        FILE* fp = __iob_func();
+        stdin = fp;
+        stdout = fp + 1;
+        stderr = fp + 1;
```
Comment (Daniel-N) [inline src/rt/stdio_msvc.c]: stderr = fp + 2;

Comment (kinke) [inline src/rt/stdio_msvc.c]: Already fixed by https://github.com/D-Programming-Language/druntime/pull/1361, i.e., in master too.

**Pull request merged**